docs/source/contributing/developing.adoc

```diff
@@ -274,12 +274,28 @@ RUN_BEFORE_FEATURE::
 The `RUN_BEFORE_FEATURE` parameter specifies {project} commands to be run before each feature.
 This provides the ability to run integration tests against a {project} instance which is not in its default state.
 When multiple commands are specified, they must be delimited by a semicolon.
-For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following:
+For example, tests can be run against a stopped {project} instance with the image caching option turned on by running the following: 
 
 ----
 $ make integration_all RUN_BEFORE_FEATURE="start; stop; config set image-caching true"
 ----
 
+TEST_WITH_SHELL::
+
+About:::
+The majority of {project} integration tests pass command line arguments directly to the `minishift` or `oc` binaries via Go's `os/exec` package.
+However, some integration tests first start an instance of the shell to pass commands into later.
+This allows testing of commands which are dependent on the shell they are being executed in.
```

Comment:
Is a single shell session maintained, or is this started on each command invocation?

If re-invoked, it means the environment variables are being reset. Right?
This would not be preferred, and in this case would need documentation (actual issue could be fixed in a later PR).

Reply:
Shell session is maintained. This should be clarified in the doucmentation, as this is important as it means setting variables and and oneliners would be possible.

Reply:
@gbraad Thought the concept and implementation of `user starts shell instance` -> `executing "docker info" in shell` -> `user closes shell instance` will talk for itself. But I agree that if there is this part, it could mention it explicitly. WDYT about:

```
About:::
The majority of {project} integration tests pass command line arguments directly to the `minishift` or `oc` binaries via Go's `os/exec` package.
However, some integration tests first start persistent instance of the shell to pass commands into later, for example: `When user starts shell instance on host machine`.
Due to persistency of the shell instance all changes done in it will be present until its closure, for example, changed environmental variables.
This allows testing of commands which are dependent on the shell they are being executed in.
The steps using this approach have the `shell` keyword in their definition, for example: `When executing "minishift oc-env" in host shell succeeds`.
```